functional-tests.sh

```diff
@@ -43,14 +43,54 @@ run unordered_bed $exe --by tests/unordered.bed t tests/ovl.bam
 assert_exit_code 0
 assert_equal $(zcat t.regions.bed.gz | wc -l) 2
 
+run bed3_n_entries $exe --by tests/bed3.bed t tests/ovl.bam
+assert_exit_code 0
+assert_equal $(zcat t.regions.bed.gz | wc -l) 2
+
+run bed4_n_entries $exe --by tests/bed4.bed t tests/ovl.bam
+assert_exit_code 0
+assert_equal $(zcat t.regions.bed.gz | wc -l) 2
+
+run bed5_n_entries $exe --by tests/bed5.bed t tests/ovl.bam
+assert_exit_code 0
+assert_equal $(zcat t.regions.bed.gz | wc -l) 2
+
+run bed6_n_entries $exe --by tests/bed6.bed t tests/ovl.bam
+assert_exit_code 0
+assert_equal $(zcat t.regions.bed.gz | wc -l) 2
+
+run bed12_n_entries $exe --by tests/bed12.bed t tests/ovl.bam
+assert_exit_code 0
+assert_equal $(zcat t.regions.bed.gz | wc -l) 2
+
+run bed3_n_columns $exe --by tests/bed3.bed t tests/ovl.bam
+assert_exit_code 0
+assert_equal $(zcat t.regions.bed.gz | awk 'BEGIN{FS="\t"}{print NF}' | sort | uniq) 5
+
+run bed4_n_columns $exe --by tests/bed4.bed t tests/ovl.bam
+assert_exit_code 0
+assert_equal $(zcat t.regions.bed.gz | awk 'BEGIN{FS="\t"}{print NF}' | sort | uniq) 5
+
+run bed5_n_columns $exe --by tests/bed5.bed t tests/ovl.bam
+assert_exit_code 0
+assert_equal $(zcat t.regions.bed.gz | awk 'BEGIN{FS="\t"}{print NF}' | sort | uniq) 5
+
+run bed6_n_columns $exe --by tests/bed6.bed t tests/ovl.bam
+assert_exit_code 0
+assert_equal $(zcat t.regions.bed.gz | awk 'BEGIN{FS="\t"}{print NF}' | sort | uniq) 6
+
+run bed12_n_columns $exe --by tests/bed12.bed t tests/ovl.bam
+assert_exit_code 0
+assert_equal $(zcat t.regions.bed.gz | awk 'BEGIN{FS="\t"}{print NF}' | sort | uniq) 12
+
 # theres data left in the bam but the region tree is empty...
 run missing_bed_chrom $exe --by tests/missing.bed t tests/ovl.bam
 assert_exit_code 0
 
 run big_window $exe t tests/ovl.bam --by 100000000
 assert_exit_code 0
 assert_equal $(zgrep -c "MT" t.per-base.bed.gz) 2
-assert_equal "MT	0	16569	0.00" "$(zgrep ^MT t.regions.bed.gz)"
+assert_equal "MT	0	16569	MT:0-16569	0.00" "$(zgrep ^MT t.regions.bed.gz)"
```

Comment:
the name column needs to be added to any existing checks or any new checks created as that column is now always present in output bed files